nevergrad/functions/photonics/photonics.py

```diff
@@ -220,3 +220,251 @@ def morpho(X: np.ndarray) -> float:
+        if g>2:
+            gamma[1:g-2]=gamma[1:g-2]*(1-2*(np.imag(gamma[1:g-2])<0))
+        if np.real(Epsilon[Type[g-1]])<0 and np.real(Mu[Type[g-1]])<0 and np.real(np.sqrt(Epsilon[Type[g-1]]*Mu[Type[g-1]]*k0**2-alpha**2))!=0:
+            gamma[g-1]=-np.sqrt(Epsilon[Type[g-1]]*Mu[Type[g-1]]*k0**2-alpha**2)
+        else :
+            gamma[g-1]=np.sqrt(Epsilon[Type[g-1]]*Mu[Type[g-1]]*k0**2-alpha**2)
+        T=np.zeros(((2*g,2,2)),dtype=complex)
+        T[0]=[[0,1],[1,0]]
+        for k2 in range(g-1):
+            t=np.exp(i*gamma[k2]*hauteur[k2])
+            T[2*k2+1]=[[0,t],[t,0]]
+        #Interface scattering matrix
+            b1=gamma[k2]/f[Type[k2]]
+            b2=gamma[k2+1]/f[Type[k2+1]]
+            T[2*k2+2]=[[(b1-b2)/(b1+b2),2*b2/(b1+b2)],[2*b1/(b1+b2),(b2-b1)/(b1+b2)]]
+            t=np.exp(i*gamma[g-1]*hauteur[g-1])
+            T[2*g-1]=[[0,t],[t,0]]
+        H=np.zeros(((2*g-1,2,2)),dtype=complex)
+        A=np.zeros(((2*g-1,2,2)),dtype=complex)
+        H[0]=T[2*g-1]
+        A[0]=T[0]
+        for j in range(2*g-2):
+            A[j+1]=cascade2(A[j],T[j+1])
+            H[j+1]=cascade2(T[2*g-2-j],H[j])
+        r=A[len(A)-1][0,0]
+        t=A[len(A)-1][1,0]
+        I=np.zeros(((2*g,2,2)),dtype=complex)
+        for j in range(len(T)-1):
+            I[j][0,0]=A[j][1,0]/(1-A[j][1,1]*H[len(T)-2-j][0,0])
+            I[j][0,1]=A[j][1,1]*H[len(T)-2-j][0,1]/(1-A[j][1,1]*H[len(T)-2-j][0,0])
+            I[j][1,0]=A[j][1,0]*H[len(T)-2-j][0,0]/(1-A[j][1,1]*H[len(T)-2-j][0,0])
+            I[j][1,1]=H[len(T)-2-j][0,1]/(1-A[j][1,1]*H[len(T)-2-j][0,0])
+        I[2*g-1][0,0]=I[2*g-2][0,0]*np.exp(i*gamma[g-1]*hauteur[g-1])
+        I[2*g-1][0,1]=I[2*g-2][0,1]*np.exp(i*gamma[g-1]*hauteur[g-1])
+        I[2*g-1][1,0]=0
+        I[2*g-1][1,1]=0
+        w=0
+        poynting=np.zeros(2*g,dtype=complex)
+        if pol==0: #TE
+            for j in range(2*g):
+                poynting[j]=np.real( (I[j][0,0]+I[j][1,0]) * np.conj( (I[j][0,0]-I[j][1,0] )*gamma[w]/Mu[Type[w]] )*Mu[Type[0]]/(gamma[0]))
+                w=w+1-np.mod(j+1,2)
+        else:       #TM
+            for j in range(2*g):
+                poynting[j]=np.real( (I[j][0,0]-I[j][1,0]) * np.conj( (I[j][0,0]+I[j][1,0] )*gamma[w]/Epsilon[Type[w]] )*Epsilon[Type[0]]/(gamma[0]))
+                w=w+1-np.mod(j+1,2)
+        tmp=abs(-np.diff(poynting))
+        absorb=tmp[np.arange(0,2*g,2)]
+        r=A[len(A)-1][0,0]
+        t=A[len(A)-1][1,0]
+        R=abs(r)**2
+        T=abs(t)**2*gamma[g-1]*f[Type[0]]/(gamma[0]*f[Type[g-1]])
+        scc[k]=solar(vlam[k])
+        Ab[k]=absorb[len(absorb)-1]
+    max_scc=np.trapz(scc,vlam)
+    j_sc=np.trapz(scc*Ab,vlam)
+    CE=j_sc/max_scc
+    cost=1-CE
+    return(cost)
+
+### test photosic_reference
+#debut=np.array([79,102])
+#fin=np.array([100,70])
+#dbr=np.tile(np.array([147,120]),3)
+#X=np.append(debut,np.append(dbr,fin))
+#cf_test=cf_photosic_reference(X)
+#print(cf_test) #0.09069397958516512
```

Comment:
this would be perfect for a test, are you familiar with unitests?

Adding something like
```python
def test_photosic_reference() -> None:
    debut=np.array([79,102])
    fin=np.array([100,70])
    dbr=np.tile(np.array([147,120]),3)
    X=np.append(debut,np.append(dbr,fin))
    cf_test=cf_photosic_reference(X)
     np.testing.assert_almost_equal(cf_test, 0.09069397)
```
in photonics/test_core.py should work and will be automatically tested at each PRs

Reply:
OK, made in test_core.py